src/components/WalletModal/index.tsx

```diff
@@ -135,48 +137,48 @@ const ModalContent = (props: any) => {
 
     const { ethereum } = window
     const toggleNetworkModal = useNetworkModalToggle()
-
+    const isMultiple = useSelectedProvider()
     const handleEthereumNetworkSwitch = useCallback(() => {
         const networkType = process.env.REACT_APP_NETWORK || 'staging'
         if (networkType === 'staging') {
             toggleNetworkModal()
         } else if (networkType === 'production') {
-            ;(ethereum as any)?.request({
+              ;(isMultiple ? ethereum?.selectedProvider as any : ethereum as any)?.request({
```

Comment:
what if it is not multiple?

Reply:
then the ethereum object is filled with only one provider and we can just ethereum as any)?.request